src/sentry/static/sentry/app/components/externalIssues/abstractExternalIssueForm.tsx

```diff
@@ -0,0 +1,244 @@
+import React from 'react';
+import {Modal} from 'react-bootstrap';
+import debounce from 'lodash/debounce';
+import * as queryString from 'query-string';
+
+import AsyncComponent from 'app/components/asyncComponent';
+import {IntegrationIssueConfig, IssueConfigField} from 'app/types';
+import FieldFromConfig from 'app/views/settings/components/forms/fieldFromConfig';
+import Form from 'app/views/settings/components/forms/form';
+import {FieldValue} from 'app/views/settings/components/forms/model';
+import {FormField} from 'app/views/settings/projectAlerts/issueEditor/ruleNode';
+
+export type ExternalIssueAction = 'create' | 'link';
+
+type Props = {
+  action?: ExternalIssueAction;
+} & AsyncComponent['props'];
+
+type State = {
+  integrationDetails?: IntegrationIssueConfig;
+  dynamicFieldValues?: {[key: string]: FieldValue};
+} & AsyncComponent['state'];
+
+const DEBOUNCE_MS = 200;
+
+/**
+ * @abstract
+ */
+export default class AbstractExternalIssueForm<
+  P extends Props = Props,
+  S extends State = State
+> extends AsyncComponent<P, S> {
+  shouldRenderBadRequests = true;
+  action: ExternalIssueAction;
+
+  constructor(props: P, context: any) {
+    super(props, context);
+    this.action = (this.props.action || 'create') as ExternalIssueAction;
+  }
+
+  getDefaultState(): State {
+    return {
+      ...super.getDefaultState(),
+      // This is derived from integrationDetails when it loads.
+      dynamicFieldValues: undefined,
+      // This is fetched by AsyncComponent.getEndpoints.
+      integrationDetails: undefined,
+    };
+  }
+
+  getEndPointString = () => '';
```

Comment:
@mgaeta I think it's better throw an error for this function not being implemented but specify the return types so it can know how to use it. See what we do in the `AbstractIntegrationDetailedView`.